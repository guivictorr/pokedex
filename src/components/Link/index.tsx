import NextLink, { LinkProps } from 'next/link';
import { ReactNode } from 'react';

type MyLinkProps = LinkProps & {
  children: ReactNode;
};

const Link = ({ children, ...rest }: MyLinkProps) => {
  return (
    <NextLink {...rest}>
      <a>{children}</a>
    </NextLink>
  );
};

export default Link;
