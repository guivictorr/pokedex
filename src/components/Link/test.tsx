import { render, screen } from '@testing-library/react';
import Link from '.';

describe('<Link/>', () => {
  it('should render correctly', () => {
    render(<Link href="https://www.google.com">Link</Link>);
    const link = screen.getByRole('link', { name: /link/i });
    expect(link).toBeInTheDocument();
    expect(link).toMatchSnapshot();
  });

  it('should render with correct href', () => {
    render(<Link href="https://www.google.com">Google</Link>);

    expect(screen.getByRole('link', { name: /google/i })).toHaveAttribute(
      'href',
      'https://www.google.com',
    );
  });
});
