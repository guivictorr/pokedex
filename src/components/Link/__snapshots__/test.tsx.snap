// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link/> should render correctly 1`] = `
<a
  href="https://www.google.com"
>
  Link
</a>
`;
